listenbrainz/webserver/static/js/src/follow/UserSocialNetwork.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from "react";
+import { includes as _includes } from "lodash";
+
+import APIService from "../APIService";
+import FollowerFollowingModal from "./FollowerFollowingModal";
+import SimilarUsersModal from "./SimilarUsersModal";
+
+export type UserSocialNetworkProps = {
+  user: ListenBrainzUser;
+  loggedInUser: ListenBrainzUser | null;
+};
+
+type UserSocialNetworkState = {
+  followerList: Array<ListenBrainzUser>;
+  followingList: Array<ListenBrainzUser>;
+  similarUsersList: Array<SimilarUser>;
+};
+
+export default class UserSocialNetwork extends React.Component<
+  UserSocialNetworkProps,
+  UserSocialNetworkState
+> {
+  APIService: APIService;
+
+  constructor(props: UserSocialNetworkProps) {
+    super(props);
+    this.APIService = new APIService(`${window.location.origin}/1`);
+    this.state = {
+      followerList: [],
+      followingList: [],
+      similarUsersList: [],
+    };
+
+    this.getFollowing();
+    this.getFollowers();
+    this.getSimilarUsers();
+  }
+
+  getSimilarUsers = () => {
+    const { user } = this.props;
+
+    this.APIService.getSimilarUsersForUser(user.name).then(
+      ({
+        payload,
+      }: {
+        payload: Array<{ user_name: string; similarity: number }>;
+      }) => {
+        this.setState({
+          similarUsersList: payload.map(
+            (similarUser: { user_name: string; similarity: number }) => {
+              return {
+                name: similarUser.user_name,
+                similarityScore: similarUser.similarity,
+              } as SimilarUser;
```

Comment:
Ah I see this is where you're transforming the shape of the similar users. I think this will be fine for now, and allow us to interchangeably pass around  `ListenBrainzUser` or `SimilarUser` objects.

However that `as SimilarUser` shouldn't be needed and can hide typescript issues.